packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -610,6 +825,66 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig: DeploymentAlarmConfig) {
+    // Throw an error if alarms array is empty
+    if (alarmConfig.alarms.length === 0) {
+      throw new Error('Specify at least one deployment alarm');
+    }
+    this.deploymentAlarms = {
+      alarmNames: alarmConfig.alarms.map(alarm => alarm.alarmName),
+      enable: true,
+      rollback: alarmConfig.behavior !== AlarmBehavior.FAIL_ON_ALARM,
+    };
+  }
+
+  /**
+   *   Disassociate existing deployment alarms
+  */
+  public disableDeploymentAlarms() {
+    if (this.deploymentAlarms) {
+      this.deploymentAlarms = {
+        alarmNames: [],
+        rollback: false,
+        enable: false,
+      };
+    }
+  }
+
+  /**
+   *   Add an alarm based on ECS metric which appended to the list of deployment alarms at synthesis time
+   *   Customize the alarm threshold and evaluationPeriods using 2nd argument ecsMetricAlarmProps.
+   *   `threshold  value 85`
+   *   `evaluationPeriods value 3`
+  */
+  public createEcsMetricAlarm(metric: EcsMetric, ecsMetricAlarmProps?: EcsMetricAlarmProps): cloudwatch.Alarm {
+    // Throw an error if service connect is not configured
+    if (metric.isServiceConnectMetric() && !this._serviceConnectConfig) {
+      throw new Error('Service connect must be enabled to set service connect metric alarms.');
+    }
+    const ecsMetric = this.metric(metric.customValue, ecsMetricAlarmProps?.metricProps);
+    const alarmName = Names.uniqueId(this);
+    const metricAlarm = new cloudwatch.Alarm(this, alarmName,
+      ecsMetricAlarmProps?.alarmProps ? ecsMetricAlarmProps.alarmProps : {
+        metric: ecsMetric,
+        threshold: ecsMetricAlarmProps?.alarmProps?.threshold || 85,
+        evaluationPeriods: ecsMetricAlarmProps?.alarmProps?.evaluationPeriods || 3,
+        alarmName,
```

Comment:
[Important]: Why are we setting this instead of just leaving it to cloudformation and using references?

Reply:
I've tried to remove this and it turns out that we need to be able to create a name for the alarm, otherwise it's super easy to create a circular dependency. 

For example: 
```typescript
const svc = new FargateService(stack, 'svc', {});

// Metrics aren't actually constructs; they're convenience objects that make it easy to create alarms
const metric = svc.metricCpuUtilization();

svc.createAlarm({
  useAsDeploymentAlarm: true,
  metric,
  evaluationPeriods: 4,
  threshold: 80,
});
// This creates a circular dependency because the metric dimensions include the service name! 
// That means that the alarm has a reference to the service, but the service has a reference to the alarm, because we have to pass the alarm name to `deploymentAlarms.alarmNames`.
```

To break this, we need deterministic string names, which unfortunately means we face a tradeoff.
1. Provide a unique id which we generate on each deployment, which re-creates the resource every single time. 
2. Use a deterministic name, which means we can't do any updates on the alarm which require replacement. Currently, `AlarmName` ([docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudwatch-alarm.html#cfn-cloudwatch-alarms-alarmname)) is the only CFN field which requires replacement, but I don't know if there's a guarantee that this will stay the case. 

I've pushed my code which has failing tests due to this reason for visibility.